es4x/src/main/java/io/reactiverse/es4x/ESVerticleFactory.java

```diff
@@ -36,7 +36,7 @@ public int order() {
     return -1;
   }
 
-  abstract Verticle createVerticle(Runtime runtime, String fsVerticleName);
+  protected  abstract Verticle createVerticle(Runtime runtime, String fsVerticleName);
```

Comment:
Makes sense.

Reply:
The main objective is to be able to embed ES4X into an application. This app need :

- to give JS scripts access to some instances (not only Java.type)
- to hook into some lifecycle callbacks (beyond start and stop, there is more related to this app)
- load scripts from other sources other than file system (remote, database, resources classpath)
- node compatibility (is not the main concern)

Reply:
To address these issues:

* to expose more instances you can do `runtime.put('variableName', anyObject)` this will be exposed using the bindings.
* to have more hooks (I assume that is why you want to change the polyfills) and load scripts from other locations (again the polyfills) perhaps we can make the polyfills configurable, instead of hardcoded, so in the constructor accept a list of `Source` objects. This way you can load them from where you want and they will be locked at runtime instantiation.

Reply:
Yes thank you! This would be perfect for me,  I added Source[] as Runtime constructor parameter and propgated to newContext method. this has been added as an optional parameter (new variants to the constructor and newContext method).